cli/command/trust/key_generate.go

```diff
@@ -0,0 +1,131 @@
+package trust
+
+import (
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary"
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/data"
+	tufutils "github.com/docker/notary/tuf/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type keyGenerateOptions struct {
+	name      string
+	directory string
+}
+
+func newKeyGenerateCommand(dockerCli command.Streams) *cobra.Command {
+	options := keyGenerateOptions{}
+	cmd := &cobra.Command{
+		Use:   "generate NAME [NAME...]",
+		Short: "Generate and load a signing key-pair",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.name = args[0]
+			return setupPassphraseAndGenerateKeys(dockerCli, options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&options.directory, "dir", "d", "", "Directory to generate key in, defaults to current directory")
```

Comment:
I removed `-d` and left `--dir`, because I thought it'd good to be prescriptive with the filenames themselves (having `.pub` extension and the specified).

I'm holding off on `stdin` for now because the env var `DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE` can be used

Reply:
I see this env variable is from previous commits, so maybe it doesn't need fixing here, but it's worth noting that this is also very unsafe. Just as `/proc/self/cmdline` has the contents of the command line arguments, `/proc/self/environ` has the contents of the task's initial environment, and is world readable, exposing you to similar issues as `/proc/self/cmdline`.